lib/backend/k8s/resources/profile.go

```diff
@@ -222,55 +224,101 @@ func (c *profileClient) EnsureInitialized() error {
 }
 
 func (c *profileClient) Watch(ctx context.Context, list model.ListInterface, revision string) (api.WatchInterface, error) {
-	resl := list.(model.ResourceListOptions)
-	if len(resl.Name) != 0 {
-		return nil, fmt.Errorf("cannot watch specific resource instance: %s", list.(model.ResourceListOptions).Name)
+	// Build watch options to pass to k8s.
+	opts := metav1.ListOptions{Watch: true}
+	rlo, ok := list.(model.ResourceListOptions)
+	if !ok {
+		return nil, fmt.Errorf("ListInterface is not a ResourceListOptions: %s", list)
+	}
+
+	// Setting to Watch all profiles in all namespaces; overriden below
+	namePrefix := ""
+	ns := kapiv1.NamespaceAll
+	sa := ""
+
+	// Watch a specific profile.
+	if len(rlo.Name) != 0 {
+		log.WithField("name", rlo.Name).Debug("Watching a single profile")
+		var err error
+		if strings.HasPrefix(rlo.Name, conversion.NamespaceProfileNamePrefix) {
+			namePrefix = conversion.NamespaceProfileNamePrefix
+			ns, err = c.ProfileNameToNamespace(rlo.Name)
+			if err != nil {
+				return nil, err
+			}
+			opts.FieldSelector = fields.OneTermEqualSelector("metadata.name", ns).String()
+		} else if strings.HasPrefix(rlo.Name, conversion.ServiceAccountProfileNamePrefix) {
+			namePrefix = conversion.ServiceAccountProfileNamePrefix
+			ns, sa, err = c.ProfileNameToServiceAccount(rlo.Name)
+			if err != nil {
+				return nil, err
+			}
+			opts.FieldSelector = fields.OneTermEqualSelector("metadata.name", sa).String()
+		} else {
+			return nil, fmt.Errorf("Unsupported prefix for resource name: %s", rlo.Name)
+		}
 	}
 
 	nsRev, saRev, err := c.SplitProfileRevision(revision)
 	if err != nil {
 		return nil, err
 	}
+	{
```

Comment:
Hm, this is a bit complicated just for a log line... I was thinking more just move the log lines right next to the point where we create the watcher - at that point we have the knowledge about whether we're starting a particular watch.

No need to maintain a single log event for this.